Documentation: Details/Notes regarding Rails API-only applications (#5152)
plataformatec/devise: README.md

### Rails API Mode

> Rails 5+ has a built-in [API Mode](https://edgeguides.rubyonrails.org/api_app.html) which optimizes Rails for use as an API (only). Devise is _somewhat_ able to handle applications that are built in this mode without additional modifications in the sense that it should not raise exceptions and the like. But some issues may still arise during `development`/`testing`, as we still don't know the full extent of this compatibility. (For more information, see #4947)

Reviewer: ```suggestion
Rails 5+ has a built-in [API Mode](https://edgeguides.rubyonrails.org/api_app.html) which optimizes Rails for use as an API (only). Devise is _somewhat_ able to handle applications that are built in this mode without additional modifications in the sense that it should not raise exceptions and the like. But some issues may still arise during `development`/`testing`, as we still don't know the full extent of this compatibility. (For more information, see [issue #4947](https://github.com/plataformatec/devise/issues/4947/))
```

#### Supported Authentication Strategies
The only default configured strategy supported by devise for API-only applications is database via HTTP Auth, meaning the user is effectively authenicated on each request (a byproduct of API-only applications not supporting browser-based sessions via cookies).

The devise default for HTTP Auth is disabled, so it will need to be enabled in the devise initializer for the database strategy:

```ruby
config.http_authenticatable = [:database]
```

> This restriction does not limit you from implementing custom strategies, either in your application or via gem-based extensions for devise.

Reviewer: I think we can be a little bit more explicit that we are talking about `warden` strategies here, something like
```suggestion
This restriction does not limit you from implementing custom warden strategies, either in your application or via gem-based extensions for devise.
```

> A common authentication strategy for APIs is token-based authenication. For more information on extending devise to support this type of authentication and others, see the wiki article for [Simple Token Authentication Examples and alternatives](https://github.com/plataformatec/devise/wiki/How-To:-Simple-Token-Authentication-Example#alternatives).

Reviewer: Little typo 😄
But I also think that we can link [@tegon's article](http://blog.plataformatec.com.br/2019/01/custom-authentication-methods-with-devise/) about adding custom authentication strategies here. Something like:

```suggestion
A common authentication strategy for APIs is token-based authentication. For more information on extending devise to support this type of authentication and others, see the wiki article for [Simple Token Authentication Examples and alternatives](https://github.com/plataformatec/devise/wiki/How-To:-Simple-Token-Authentication-Example#alternatives) or the this blog post on [Custom authentication methods with Devise](http://blog.plataformatec.com.br/2019/01/custom-authentication-methods-with-devise/).
```

> A common authentication strategy for APIs is token-based authentication. For more information on extending devise to support this type of authentication and others, see the wiki article for [Simple Token Authentication Examples and alternatives](https://github.com/plataformatec/devise/wiki/How-To:-Simple-Token-Authentication-Example#alternatives) or the this blog post on [Custom authentication methods with Devise](http://blog.plataformatec.com.br/2019/01/custom-authentication-methods-with-devise/).

Reviewer: ```suggestion
A common authentication strategy for APIs is token-based authentication. For more information on extending devise to support this type of authentication and others, see the wiki article for [Simple Token Authentication Examples and alternatives](https://github.com/plataformatec/devise/wiki/How-To:-Simple-Token-Authentication-Example#alternatives) or this blog post on [Custom authentication methods with Devise](http://blog.plataformatec.com.br/2019/01/custom-authentication-methods-with-devise/).
```
